### PR TITLE
[BEAM-25] Temporarily reject stateful ParDo in DataflowRunner (until support is added)

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

R: @bjchambers   

At present, the user-facing API for stateful `DoFn` is in the codebase but prevented from use. It is `ParDo.of(...)` that rejects stateful `DoFn`. In #1399 I am removing this rejection, so I need to add it to all runners until they support the API.

I have also added a JUnit category so runners can exclude this from their `RunnableOnService` test suites.

### CODE REVIEW

**Comment by kennknowles** [comment]
Filed [BEAM-1038](https://issues.apache.org/jira/browse/BEAM-1038) against myself following @amitsela's comments on #1409.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java [hunk 1]
```diff
@@ -959,6 +961,18 @@ public void translate(
           private <InputT, OutputT> void translateMultiHelper(
               ParDo.BoundMulti<InputT, OutputT> transform,
               TranslationContext context) {
+
+            DoFn<InputT, OutputT> doFn = transform.getNewFn();
+            if (DoFnSignatures.getSignature(doFn.getClass()).stateDeclarations().size() > 0) {
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java]: Might be nicer if there was a "DoFnSignatures.getSignature(...).isStateful()" or something to easily categorize these?

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java [hunk 2]
```diff
@@ -987,6 +1001,17 @@ public void translate(
           private <InputT, OutputT> void translateSingleHelper(
               ParDo.Bound<InputT, OutputT> transform,
               TranslationContext context) {
+            DoFn<InputT, OutputT> doFn = transform.getNewFn();
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java]: Why not introduce a "rejectStatefulParDo(...)" helper and share this code?

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java (hunk 1 above)
Comment (kennknowles) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java]: Done

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java (hunk 2 above)
Comment (kennknowles) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java]: Done

**Comment by kennknowles** [comment]
I'll reorder the change to `DoFnSignature` before the other changes when I squash.

**Pull request closed without merging**